Update default filesystem to use root of system path

bryannielsen opened the pull request:
## Overview

Update the path used by default filesystem instance to be the root of SYSPATH.  This is more in line with what previous versions of ExpressionEngine allowed and should resolve issues where an installation is spread out across a volume.

Resolves EECORE-2093

## Nature of This Change

<!-- Check all that apply: -->

- [x] 🐛 Fixes a bug
- [ ] 🚀 Implements a new feature
- [ ] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [ ] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security <!-- if your fix would EXPOSE a current security flaw, do not submit a pull request. Instead report a security bug at https://docs.expressionengine.com/latest/bugs_and_security_reports -->

## Is this backwards compatible?

- [x] Yes
- [ ] No

bryannielsen committed:
Update default filesystem to use root of system path

EECORE-2093

bryannielsen committed:
Fix upload writable check to use upload_path

intoeetive reviewed (dismissed):
I have tested this with file manager, templates and 1-click update using various system folder locations, and it seems to work fine on Windows.

Approving, but I think we need more testing and approving here.

intoeetive commented:
Looks like this does not solve some of possibly related issues: https://github.com/ExpressionEngine/ExpressionEngine/issues/2303#issuecomment-1222199606

Yulyaswan commented:
I've tested file manager: create a new directory, add folders, move files between folders and directories, edit, delete, copy link, drag and drop files, and add a new file using the `upload new` button. Everything seems to be working.

bryannielsen committed:
Fix upload to not follow symlink out of destination

Previously using symlink path in your Upload Destination's server_path
could make it impossible to upload a file

bryannielsen committed:
Throw exception when createTempFile() fails

matthewjohns0n reviewed (approved):
I tested this on my site that is set up with a separate private path and this is now working.
